otelcol/command_examine_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelcol // import "go.opentelemetry.io/collector/otelcol"
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+func TestExamineCommand(t *testing.T) {
+	tests := []struct {
+		name      string
+		set       confmap.ResolverSettings
+		errString string
+	}{
+		{
+			name:      "no URIs",
+			set:       confmap.ResolverSettings{},
+			errString: "at least one config flag must be provided",
+		},
+		{
+			name: "valid URI - file not found",
+			set: confmap.ResolverSettings{
+				URIs: []string{"file:blabla.yaml"},
+				ProviderFactories: []confmap.ProviderFactory{
+					fileprovider.NewFactory(),
+				},
+				DefaultScheme: "file",
+			},
+			errString: "cannot retrieve the configuration: unable to read the file",
+		},
+		{
+			name: "valid URI",
+			set: confmap.ResolverSettings{
+				URIs: []string{"yaml:processors::batch/foo::timeout: 3s"},
+				ProviderFactories: []confmap.ProviderFactory{
+					yamlprovider.NewFactory(),
+				},
+				DefaultScheme: "yaml",
+			},
+		},
+		{
+			name: "valid URI - no provider set",
+			set: confmap.ResolverSettings{
+				URIs:          []string{"yaml:processors::batch/foo::timeout: 3s"},
+				DefaultScheme: "yaml",
+			},
+			errString: "at least one Provider must be supplied",
+		},
```

Comment:
I was expecting a test with at least a single config source and an assertion that the yaml configuration matches the source.

Reply:
Yup, that would be a great addition. I'll do that. Thanks!